cpp/src/groupby/sort/groupby.cu

```diff
@@ -403,6 +403,11 @@ void store_result_functor::operator()<aggregation::NTH_ELEMENT>(aggregation cons
 template <>
 void store_result_functor::operator()<aggregation::COLLECT>(aggregation const& agg)
 {
+  auto null_handling =
+    static_cast<cudf::detail::collect_list_aggregation const&>(agg)._null_handling;
+  CUDF_EXPECTS(null_handling == null_policy::INCLUDE,
```

Comment:
If we are supporting `null_policy::EXCLUDE`, why is this expectation?

Reply:
Hey, @rgsl888prabhu. Good question. 

Null exclusion has not been implemented for `COLLECT()` in general `groupby` aggregations. This PR specifically adds support for  `rolling_window()`.

This assertion is protection against `COLLECT()` being called on non-`rolling_window()` context with `null_policy::EXCLUDE`. The caller might otherwise get the incorrect output.

This assertion will be removed when we add support for `groupby`. Spark will certainly need this.